addons/product/pricelist.py

```diff
@@ -453,23 +290,48 @@ def _check_margin(self, cr, uid, ids, context=None):
                 return False
         return True
 
+    def _compute_pricelist_name(self, cr, uid, ids, name, args, context=None):
+        for record in self.browse(cr, uid, ids, context=context):
+            if record.compute_price == 'fixed':
+                res = ("%s %s") % (record.fixed_price, record.pricelist_id.currency_id.name)
+            elif record.compute_price == 'percentage':
+                res = _("%s %% discount") % (record.percent_price)
+            else:
+                res = _("%s %% discount and %s surcharge") % (abs(record.price_discount), record.price_surcharge)
+        return res
+
+    def _price_list_item_name(self, cr, uid, ids, name, args, context=None):
+        res = {}
+        for record in self.browse(cr, uid, ids, context=context):
+            res[record.id] = {'name': '', 'price_name': ''}
+            if record.categ_id:
+                res[record.id]['name'] = _("Category: %s") % (record.categ_id.name)
+            elif record.product_tmpl_id:
+                res[record.id]['name'] = record.product_tmpl_id.name
+            elif record.product_id:
+                product_name = record.product_id.display_name
+                product_name = product_name.replace('[%s]' % record.product_id.code, '')
```

Comment:
directly use replace in display_name. like this
record.product_id.display_name.replace('[%s]' % record.product_id.code, '')